euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -352,19 +391,58 @@ public boolean isCombinable() {
     return DAG.of(reduceState);
   }
 
+  static <VALUE> ReduceFunctor<VALUE, VALUE> toReduceFunctor(
+      CombinableReduceFunction<VALUE> reducer1) {
+
+    return new ReduceFunctor<VALUE, VALUE>() {
+      @Override
+      public boolean isCombinable() {
+        return true;
+      }
+
+      @Override
+      public void apply(Iterable<VALUE> elem, Context<VALUE> context) {
+        context.collect(reducer1.apply(elem));
+      }
+    };
+  }
+
+
+  public static class SingleValueContext<T> implements Context<T> {
+
+    T value;
+
+    @Override
+    public void collect(T elem) {
+      value = elem;
+    }
+
+    @Override
+    public Object getWindow() {
+      throw new UnsupportedOperationException(
+          "The window is unknown when applying combinable function");
```

Comment:
* regarding the exception: that's true! i overlooked that. (assuming users never implement their own `ReduceFunctor` with an overridden `isCombinable`.) but then i don't really like it :( but i can live with it :) no need to change it. btw: is there a way for the executor to grasp the `CombinableReduceFunction` directly and be based directly on that?
* i'm okey with the duplication and also okey with the `util` package.